app/controllers/user_controller.rb

```diff
@@ -0,0 +1,51 @@
+ServiceDiscovery::App.controllers :users do
+  
+  post '/:user_id/address' do
+    get_create_address_params
+    address = create_address @input, params[:user_id]
+    status 201
+    {address_id: address.id}.to_json
+  end
+
+  get '/:user_id/address' do
+    addresses = Address.where('user_id' => params[:user_id])
+    status 201
+    {addresses: addresses}.to_json
+  end
+
+  put '/:user_id/update_address' do
+    get_update_address_params
+    user_id = get_valid_user_id params[:user_id]
```

Comment:
The point I am making is that keep the functionality of this method only upto validation. This method doesnt need to return user_id. If there is a validation failure, an exception will be raised